internal/provisioner/controller/gateway.go

```diff
@@ -236,6 +236,20 @@ func (r *gatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	if gatewayClassParams != nil {
 		contourModel.Spec.RuntimeSettings = gatewayClassParams.Spec.RuntimeSettings
 
+		if contourModel.Spec.RuntimeSettings != nil &&
+			contourModel.Spec.RuntimeSettings.Envoy != nil &&
+			contourModel.Spec.RuntimeSettings.Envoy.Metrics != nil &&
+			contourModel.Spec.RuntimeSettings.Envoy.Metrics.Port != 0 {
+
+			port := contourModel.Spec.RuntimeSettings.Envoy.Metrics.Port
+			contourModel.Spec.NetworkPublishing.Envoy.Ports = append(
+				contourModel.Spec.NetworkPublishing.Envoy.Ports,
+				model.Port{
+					Name:          "metrics",
+					ServicePort:   int32(port),
+					ContainerPort: int32(port),
+				})
```

Comment:
This has the effect of exposing the metrics port on the Service, which we don't do when using the default port and I don't think we want to do for custom ports. Were you intentionally trying to expose it on the Service? If so, could explain the thinking further